core/autocrypt/crypto.py

```diff
@@ -0,0 +1,378 @@
+# -*- coding: utf-8 -*-
+# vim:ts=4:sw=4:expandtab
+"""Crypto implements the OpenPGP operations needed for Autocrypt.
+The API is the same as in bingpg.py.
+"""
+
+from __future__ import print_function, unicode_literals
+
+import glob
+import os
+import re
+import sys
+
+import six
+from pgpy import PGPUID, PGPKey, PGPMessage, PGPKeyring, PGPSignature
+from pgpy.constants import (CompressionAlgorithm, HashAlgorithm, KeyFlags,
+                            PubKeyAlgorithm, SymmetricKeyAlgorithm)
+
+# TODO: these two functions should be in a separate file
+from .bingpg import KeyInfo, b64encode_u
+
+
+# NOTE: key size was decided to be 2048
+KEY_SIZE = 2048
+# TODO: see which defaults we would like here
+SKEY_ARGS = {
+    'hashes': [HashAlgorithm.SHA512, HashAlgorithm.SHA256],
+    'ciphers': [SymmetricKeyAlgorithm.AES256,
+                SymmetricKeyAlgorithm.AES192,
+                SymmetricKeyAlgorithm.AES128],
+    'compression': [CompressionAlgorithm.ZLIB,
+                    CompressionAlgorithm.BZ2,
+                    CompressionAlgorithm.ZIP,
+                    CompressionAlgorithm.Uncompressed]
+}
+# RSAEncrypt is deprecated, therefore using RSAEncryptOrSign
+# also for the subkey
+SKEY_ALG = PubKeyAlgorithm.RSAEncryptOrSign
+SKEY_USAGE_SIGN = {KeyFlags.Sign}
+SKEY_USAGE_ENC = {KeyFlags.EncryptCommunications, KeyFlags.EncryptStorage}
+SKEY_USAGE_ALL = {KeyFlags.Sign, KeyFlags.EncryptCommunications,
+                  KeyFlags.EncryptStorage}
+
+
+def key_bytes(pgpykey):
+    """Key bytes.
+
+    :param key: key (either public or private)
+    :type key: PGPKey
+    :return: key bytes
+    :rtype: string
+
+    """
+    assert isinstance(pgpykey, PGPKey)
+    if sys.version_info >= (3, 0):
+        keybytes = bytes(pgpykey)
+    else:
+        keybytes = pgpykey.__bytes__()
+    return keybytes
+
+
+class Crypto(object):
```

Comment:
what about naming it OpenPGP or PGPyAutocrypt or PGPyGPG?. Just to don't name it the same way the module is used.